services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/Hakemusalue.kt

```diff
@@ -39,7 +39,7 @@ data class KaivuilmoitusAlue(
 data class Tyoalue(
     val geometry: Polygon,
     val area: Double,
-    val tormaystarkasteluTulos: TormaystarkasteluTulos?,
+    var tormaystarkasteluTulos: TormaystarkasteluTulos?,
```

Comment:
I wish there were a way to avoid making this class mutable. The change is reflected down the class tree. This is part of the application entity, domain, request, and response data. With this change, we can no longer guarantee the immutability of any of those.

The impact of making the class mutable would be lower if the entity data had separate classes for areas.

Using `.copy()` starting from the HakemusEntityData root is not really viable since that would be messy.

If the tormaystarkastelu could be added in `KaivuilmoitusUpdateRequest.toEntityData()`, copying the areas with modifications would be doable, but we can't call a stateful service from a data class method. All the mapping logic would need to be moved to a new mapper service.

I'm not really sure what would be the best approach, or if we need to make any changes.

Reply:
I reverted this to be `val` again and implemented the update via copy-constructors.